### PR TITLE
Add Katib ROADMAP 2022/2023

### PR DESCRIPTION
**Pull request by andreyvelich**
I created initial draft for Katib ROADMAP 2022/2023.
I tried to consolidated features that we implemented in 2022, since we didn't update the ROADMAP in 2022 (if I missed some significant features that we've done, please let me know).

Please let me know if you have better ideas how to name headers.

Let's discuss the items during next month and finalise the ROADMAP.

/cc @johnugeorge @gaocegege @tenzen-y @anencore94 @kimwnasptd @elenzio9 @orfeas-k @apo-ger @d-gol @shaowei-su @zhixian82 @fischor @keisuke-umezawa @jbottum @DnPlas @juliusvonkohout 

cc @kubeflow/wg-training-leads

### CODE REVIEW

**Review by tenzen-y — COMMENTED** [review]
Thank you for compiling our roadmap!

Also, I'm thinking of creating a separate CRD for the Nas might be better. Since currently Experiment CRD seems to be a bit complex.

**Comment by andreyvelich** [comment]
>Also, I'm thinking of creating a separate CRD for the Nas might be better. Since currently Experiment CRD seems to be a bit complex.

@tenzen-y Yes, we were discussing about this before, how to separate various AutoML features from the Kubernetes CRD perspective:
1. HyperParameter tuning
2. NAS
3. Auto Feature Engineering 
4. Auto Model Compression

Then, it will be easier to maintain APIs for every aspect of AutoML.
I think, we need to decide on APIs structure before moving Katib Component to `v1` version.

**Comment by johnugeorge** [comment]
> > Also, I'm thinking of creating a separate CRD for the Nas might be better. Since currently Experiment CRD seems to be a bit complex.
> 
> @tenzen-y Yes, we were discussing about this before, how to separate various AutoML features from the Kubernetes CRD perspective:
> 
> 1. HyperParameter tuning
> 2. NAS
> 3. Auto Feature Engineering
> 4. Auto Model Compression
> 
> Then, it will be easier to maintain APIs for every aspect of AutoML. I think, we need to decide on APIs structure before moving Katib Component to `v1` version.

I feel, we need to make API changes in this release if we are planning to do so. 
Example: Can "Experiment" resource always refer to HP tuning? Should we rename the  API resource?  One other option is to add explicity resource names to other AutoML features

**Comment by tenzen-y** [comment]
> I feel, we need to make API changes in this release if we are planning to do so.

If we can collect contributors to do that, we can do that in this release. However, changing the API name is so significant work...

**Comment by gaocegege** [comment]
Should we merge this?

**Comment by andreyvelich** [comment]
Sure, I removed Katib CRDs naming (Experiment, Suggestion, Trial) from the ROADMAP, since we need to discuss what naming we are going to have in our V1 APIs version.
We discussed this [initially during our latest community call.](https://docs.google.com/document/d/1MChKfzrKAeFRtYqypFbMXL6ZIc_OgijjkvbqmwRV-64/edit#heading=h.l6zo8t476rrb)
I will create an issue for this soon.

If other items look good, we can merge it.
/assign @gaocegege @tenzen-y @johnugeorge

**Comment by tenzen-y** [comment]
Thanks Andrey!
/lgtm

File: .github/workflows/template-setup-e2e-test/action.yaml [hunk 1]
```diff
@@ -39,11 +39,11 @@ runs:
     - name: Setup Minikube Cluster
       uses: medyagh/setup-minikube@v0.0.14
       with:
-        network-plugin: cni
-        cni: flannel
+        # network-plugin: cni
+        # cni: flannel
```
Comment (tenzen-y) [inline .github/workflows/template-setup-e2e-test/action.yaml:43]: We need to set up CNI due to https://github.com/kubernetes/minikube/pull/16143.

**Review by terrytangyuan — APPROVED** [review]
/lgtm

File: .github/workflows/template-setup-e2e-test/action.yaml (hunk 1 above)
Comment (andreyvelich) [reply .github/workflows/template-setup-e2e-test/action.yaml]: Alright 22.04 work, let me submit separate PR to update the ubuntu version for our E2Es.

Comment (tenzen-y) [reply .github/workflows/template-setup-e2e-test/action.yaml]: Thank you 🎉

**Pull request merged**